### PR TITLE
add data renames (blocks & items)

### PR DESCRIPTION
**Pull request by zardoy**
fixes https://github.com/PrismarineJS/minecraft-data/issues/833

### CODE REVIEW

**Comment by rom1504** [comment]
how did you generate it ?

**Comment by zardoy** [comment]
For most versions I specified it manually (or almost manually by looking at data fixer code) since there are not many renames. But for the 1.13 version I used this script:

```js
//@ts-check
import fs from 'fs'
import minecraftData from 'minecraft-data'

const mcData = minecraftData('1.12.2')

const file = JSON.parse(fs.readFileSync('node_modules/minecraft-data/minecraft-data/data/pc/common/legacy.json', 'utf8'))

let map = []
for (const [key, item] of Object.entries(file.blocks)) {
    const item2 = mcData.blocks[key.split(':')[0]];
    if (!item2) continue
    const oldName = item2.name
    const newName = item.split(':')[1].split('[')[0];
    if (map.find(x => x[1] === newName && x[0] === oldName)) {
        continue
    }
    if (newName === oldName) {
        continue
    }
    map.push([oldName, newName])
}

fs.writeFileSync('map.json', JSON.stringify(map, null, 2))

```

**Comment by zardoy** [comment]
Im not sure if thats possible to generate with the server. I tried to extract using text manipulation (by looking at `createRenamer` in DataFixers, but I couldn't get all renames in that way). However, does the current format / file look good?

**Comment by zardoy** [comment]
Waiting for this pr, then add a function to use it to node wrapper

**Comment by rom1504** [comment]
I'm looking at the file and actually I don't understand what this is.

What is a new name ? What is an old name ?

**Comment by zardoy** [comment]
> What is a new name ? What is an old name ?

Also, the code above mentions the process of creating the map for 1.13: `map.push([oldName, newName])`. Why array? Because you might also want to go from a new name to an old name (like also mentioned in docs).

I will also open a pr in nmd so you can better understand how real code works with this format.

**Comment by rom1504** [comment]
https://github.com/PrismarineJS/node-minecraft-data/pull/335#issuecomment-1902543462 following of the conversation there.

TLDR: most of these "renames" are not renames but actually just some block being removed in old version and some block being added in new version.

A better solution may be to create "group of similar blocks" like grass, chest,... That you could handle in a similar way in downstream code

**Comment by rom1504** [comment]
The renames in this PR still contain many renames between blocks that do not have the same semantic

**Comment by rom1504** [comment]
Re open if you want to finish it
